internal/pkg/aws/s3/s3.go

```diff
@@ -251,6 +251,40 @@ func (s *S3) GetBucketTree(bucket string) (string, error) {
 	return tree.String(), nil
 }
 
+// GetBucketSizeAndCount returns the total size and number of objects in an S3 bucket.
+func (s *S3) GetBucketSizeAndCount(bucket string) (string, int, error) {
```

Comment:
The fact that this public function is specifically designed for status use case makes me think maybe we could rename this function. Otherwise it'll be too coupled with each other and not generic enough.

Also by looking at `GetBucketTree` I realized they share many very similar logics with each other (aka listing objects within the bucket). How about we just have 

```golang
func (*S3) ListObjects(bucket string) (Objects, error)

type Objects []*s3.Object

func (Objects) TotalObjectCount() int

func (Objects) TotalSize() string

func (Objects) Tree() (string, error)
```

Reply:
Hm, yeah. I hadn't reused `GetBucketTree` because of the delimiter difference, but I could pass that in. Will refactor!

Reply:
I kept total count and total size together (for now) to avoid making another API call.
I have `ListObjects` returning the output because Objects are just one element of the response that we need (we also need prefixes for tree-making).